lib/tools/adb-commands.js

```diff
@@ -190,6 +190,17 @@ methods.getReqPermissions = async function (pkg, cmdOutput = null) {
   return match[0].match(/android\.permission\.\w+/g) || [];
 };
 
+methods.getLocationProviders = async function () {
+  let stdout = await this.shell(['settings', 'get', 'secure', 'location_providers_allowed']);
+  return stdout.trim().split(',')
+    .filter((p) => !!p)
```

Comment:
I'd probably trim _before_ filtering out empties.

Another trick for filtering out empties: `filter(Boolean)` :-)